cpp/augment_bible_references.cc

```diff
@@ -294,48 +311,36 @@ bool GetBibleRanges(const std::string &field_tag, const MarcUtil::Record &record
             continue;
         }
 
-        std::vector<std::string> numbered_books;
-        CreateNumberedBooks(book_name_candidate, n_subfield_values, &numbered_books);
-        if (not HaveBibleBookCodes(numbered_books, bible_book_to_code_map)) {
+        std::vector<std::string> books;
+        CreateNumberedBooks(book_name_candidate, &n_subfield_values, &books);
+        if (not HaveBibleBookCodes(books, bible_book_to_code_map)) {
             std::cerr << fields[0] << ": found no bible book code for \"" << book_name_candidate
                       << "\"! (" << StringUtil::Join(n_subfield_values, ", ") << ")\n";
             continue;
         }
 
         std::vector<std::string> book_codes;
-        if (not ConvertNumberedBooksToBookCodes(numbered_books, bible_book_to_code_map, &book_codes)) {
+        if (not ConvertBooksToBookCodes(books, bible_book_to_code_map, &book_codes)) {
             std::cerr << fields[0] << ": can't convert one or more of these books to book codes: "
-                      << StringUtil::Join(numbered_books, ", ") << "!\n";
+                      << StringUtil::Join(books, ", ") << "!\n";
             continue;
         }
 
-        if (n_subfield_values.size() == 2) {
-            if (book_codes.size() != 1)  {
-                std::cerr << fields[0] << ": this should never happen: "
-                          << "n_subfield_values.size() == 2 AND book_codes.size() != 1\n";
-                continue;   
-            }
-            if (not ParseBibleReference(n_subfield_values[1], book_codes[0], ranges)) {
-                std::cerr << fields[0] << ": failed to parse bible references (1): " << n_subfield_values[1] << '\n';
-                continue;
-            }
-        } else if (book_codes.size() == 1) {
-            if (n_subfield_values.empty() or StringUtil::IsDigit(book_codes[0][0]))
+        if (book_codes.size() > 1) {
```

Comment:
if (book_codes.size() > 1 or n_subfield_values.empty())
	ranges->insert(std::make_pair(book_codes.first() + "00000", book_codes.last() + "99999"));
else {
    if (not ParseBibleReference(n_subfield_values.front(), book_codes[0], ranges)) {
        std::cerr << fields[0] << ": failed to parse bible references (1): " << n_subfield_values.front() << '\n';
        continue;
    }
}